File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/AggregatkennungDeserialisierer.java
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de;

import java.io.IOException;
import java.util.UUID;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer;

public class AggregatkennungDeserialisierer extends StdScalarDeserializer<Aggregatkennung>
{
    public AggregatkennungDeserialisierer() {
        super(Aggregatkennung.class);

    }

    public AggregatkennungDeserialisierer(final Class<?> vc) {
        super(vc);
    }

    @Override
    public Aggregatkennung deserialize(
            final JsonParser jsonParser,
            final DeserializationContext deserializationContext) throws IOException {

        final String uuid = jsonParser.getText();

        return Aggregatkennung.of(UUID.fromString(uuid));
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/AggregatkennungDeserialisierer.kt
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de

import java.io.IOException
import java.util.UUID

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer

class AggregatkennungDeserialisierer : StdScalarDeserializer<Aggregatkennung> {
    constructor() : super(Aggregatkennung::class.java) {

    }

    constructor(vc: Class<*>) : super(vc) {}

    @Throws(IOException::class)
    override fun deserialize(
            jsonParser: JsonParser,
            deserializationContext: DeserializationContext): Aggregatkennung {

        val uuid = jsonParser.text

        return Aggregatkennung(UUID.fromString(uuid))
    }
}